File: opt-engine/core/core-impl/src/main/java/dev/aws/proto/core/Order.java
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package dev.aws.proto.core;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Objects;

@Data
@AllArgsConstructor
public class Order {

    private String orderNo;
    private String orderDate;
    private String deliveryCode;
    private double sumWeight;

    @Override
    public int hashCode() {
        return Objects.hashCode(this.orderNo);
    }
}

File: apps_web/src/models/order.ts
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

export interface OrderData {
  Id: string
  createdAt: number
  orderNo: string
  orderDate: string
  deliveryCode: string
  deliveryName: string
  warehouseCode: string
  sumWeight: number
}

export const EMPTY_ORDER_DATA: OrderData = {
  Id: '',
  createdAt: Date.now(),
  orderNo: '',
  orderDate: '',
  deliveryCode: '',
  deliveryName: '',
  warehouseCode: '',
  sumWeight: 0,
}
